termwiz/src/tmux_cc/mod.rs

```diff
@@ -902,42 +909,40 @@ impl Parser {
         if self.buffer.last() == Some(&b'\r') {
             self.buffer.pop();
         }
-        let result = match std::str::from_utf8(&self.buffer) {
-            Ok(line) => {
+        if self.begun.is_some() {
+            return self.process_guarded_line();
+        }
+
+        let result = match parse_line(&self.buffer) {
+            Ok(Event::Begin {
+                timestamp,
+                number,
+                flags,
+            }) => {
                 if self.begun.is_some() {
-                    let line = line.to_owned();
-                    return self.process_guarded_line(line);
-                }
-                match parse_line(line) {
-                    Ok(Event::Begin {
-                        timestamp,
-                        number,
-                        flags,
-                    }) => {
-                        if self.begun.is_some() {
-                            log::error!("expected %end or %error before %begin ({})", line);
-                        }
-                        self.begun.replace(Guarded {
-                            timestamp,
-                            number,
-                            flags,
-                            error: false,
-                            output: String::new(),
-                        });
-                        None
-                    }
-                    Ok(event) => Some(event),
-                    Err(err) => {
-                        log::error!("Unrecognized tmux cc line: {}", err);
-                        return Err(anyhow::anyhow!(line.to_owned()));
-                    }
+                    log::error!(
+                        "expected %end or %error before %begin ({})",
+                        String::from_utf8_lossy(&self.buffer).as_ref()
```

Comment:
I think this can simplify like this?

```suggestion
                        String::from_utf8_lossy(&self.buffer)
```